freeipa.spec.in

```diff
@@ -1323,6 +1331,45 @@ fi
 # with_selinux
 %endif
 
+%triggerin client -- sssd-common < 2.10
+# Has the client been configured?
+restore=0
+test -f '/var/lib/ipa-client/sysrestore/sysrestore.index' && restore=$(wc -l '/var/lib/ipa-client/sysrestore/sysrestore.index' | awk '{print $1}')
+
+SSH_CLIENT_SYSTEM_CONF="/etc/ssh/ssh_config"
+if [ -f "$SSH_CLIENT_SYSTEM_CONF" ]; then
+    # https://pagure.io/freeipa/issue/9536
+    # downgrade sss_ssh_knownhosts with sss_ssh_knownhostsproxy
+    if [ -f '/usr/bin/sss_ssh_knownhosts' ]; then
+        if grep -E -q 'Include' $SSH_CLIENT_SYSTEM_CONF  2>/dev/null ; then
+            SSH_CLIENT_SYSTEM_CONF="/etc/ssh/ssh_config.d/04-ipa.conf"
+        fi
+        GLOBALKNOWNHOSTFILE="GlobalKnownHostsFile /var/lib/sss/pubconf/known_hosts/"
+        grep -qF ($GLOBALKNOWNHOSTFILE) $SSH_CLIENT_SYSTEM_CONF
+        if [ $? -ne 0 ]; then
+            sed -E --in-place=.orig '/(# IPA-related configuration changes to ssh_config)/a # added by ipa-client update\n'"$GLOBALKNOWNHOSTFILE"'' $SSH_CLIENT_SYSTEM_CONF
+        fi
+        sed -E --in-place=.orig 's/(KnownHostsCommand \/usr\/bin\/sss_ssh_knownhosts \%H)/ProxyCommand \/usr\/bin\/sss_ssh_knownhostsproxy -p \%p \%h/' $SSH_CLIENT_SYSTEM_CONF
+    fi
+fi
+
+%triggerin client -- sssd-common >= 2.10
+# Has the client been configured?
+restore=0
```

Comment:
This variable is = 0 if client packages are installed but the host is not enrolled and =2 if the host is enrolled. It should be used as a condition to execute the replacement (if the client is not installed, skip). Otherwise the upgrade of freeipa-client package triggers the following error message:
```
  Upgrading        : freeipa-client-4.12.0.dev-0.fc39.x86_64                                                                                                                5/10 
  Running scriptlet: freeipa-client-4.12.0.dev-0.fc39.x86_64                                                                                                                5/10 
sed: can't read /etc/ssh/ssh_config.d/04-ipa.conf: No such file or directory
sed: can't read /etc/ssh/ssh_config.d/04-ipa.conf: No such file or directory
warning: %triggerin(freeipa-client-4.12.0.dev-0.fc39.x86_64) scriptlet failed, exit status 2

Error in <unknown> scriptlet in rpm package freeipa-client
```

Reply:
thanks @flo-renaud , fixed.